lib/widgets/action_sheet.dart

```diff
@@ -317,3 +319,50 @@ class CopyButton extends MessageActionSheetMenuItemButton {
       data: ClipboardData(text: rawContent));
   };
 }
+
+class StarButton extends MessageActionSheetMenuItemButton {
+  StarButton({
+    super.key,
+    required super.message,
+    required super.messageListContext,
+  });
+
+  @override get icon => ZulipIcons.star;
+
+  @override
+  String label(ZulipLocalizations zulipLocalizations) {
+    return message.flags.contains(MessageFlag.starred)
+      ? zulipLocalizations.actionSheetOptionUnstarMessage
+      : zulipLocalizations.actionSheetOptionStarMessage;
+  }
+
+  @override get onPressed => (BuildContext context) async {
+    Navigator.of(context).pop();
+    String? errorMessage;
+    final op = (message.flags.contains(MessageFlag.starred))
+      ? UpdateMessageFlagsOp.remove
+      : UpdateMessageFlagsOp.add;
+    try {
+      final connection = PerAccountStoreWidget.of(messageListContext).connection;
+      await updateMessageFlags(connection, messages: [message.id],
+        op: op, flag: MessageFlag.starred);
+    } catch (e) {
+      if (!messageListContext.mounted) return;
+      final zulipLocalizations = ZulipLocalizations.of(messageListContext);
```

Comment:
nit: move this down next to the `showErrorDialog` call where it's used

(just before used, or at the top of the function, can both work well for something like this; but here feels like an odd hybrid)